native/tools/swig-config/README.md

```diff
@@ -1,379 +1,83 @@
-  if (child) {
-      console.log('child.intValue=' + child.intValue);
-      console.log('child.enabled=' + child.enabled);
-  } else {
-      console.error('child is invalid: ' + child);
-  }
+  - Create the JS configruation file, e.g.  `/Users/abc/my-project/tools/swig-config/swig-config.js` with the following content
+    
+    ```js
+    'use strict';
+    const path = require('path');
+    
+    // Developer's custom module configuration
+    // configList is required
+    const configList = [
+        [ 'your_module_interface_0.i', 'jsb_your_module_interface_0_auto.cpp' ],
+        [ 'your_module_interface_1.i', 'jsb_your_module_interface_1_auto.cpp' ],
+        // ......
+    ];
+    
+    const projectRoot = path.resolve(path.join(__dirname, '..', '..'));
+    // interfaceDir is optional
+    const interfacesDir = path.join(projectRoot, 'tools', 'swig-config');
+    // bindingsOutDir is optional
+    const bindingsOutDir = path.join(projectRoot, 'native', 'engine', 'common', 'Classes', 'bindings', 'auto');
+    
+    module.exports = {
+        interfacesDir, // optional, if it isn't exported, the items in configList should be absolute or relative to current directory of swig-config.js
+        bindingsOutDir, // optional, if it isn't exported, the items in configList should be absolute or relative to current directory of swig-config.js
+        configList // required
+    };
+    ```
   
-  const moduleTest = new my_ns.TestMyModuleSpecificMethod();
-  console.log('getModuleSpecificClass: ' + moduleTest.getModuleSpecificClass());
-  ```
-
-- Re-run project, if it succeeds, you will get the following output:
-
-  ```
-  fooBase: 111, 0
-  fooSub: hello foo sub
-  methodBeforeRenamed: 334, 564
-  17:25:10 [DEBUG]: D/ JS: public property: 123
-  17:25:10 [DEBUG]: D/ JS: attribute int value: 100
-  17:25:10 [DEBUG]: D/ JS: attribute enabled: true
-  17:25:10 [DEBUG]: D/ JS: [ERROR]: child is invalid: undefined
-  17:25:10 [DEBUG]: D/ JS: getModuleSpecificClass: undefined
-  ```
+  - Run the following command
+    
+    ```bash
+    # If current workspace is not in '/Users/abc/my-project/tools/swig-config'
+    $ node < Engine Root Path >/native/tools/swig-config/genbindings.js -c /Users/abc/my-project/tools/swig-config/swig-config.js
+    ```
+    
+    ```bash
+    # If you have already navigate to '/Users/abc/my-project/tools/swig-config' directory, you could run the command without -c argument like:
+    $ cd /Users/abc/my-project/tools/swig-config
+    $ node < Engine Root Path >/native/tools/swig-config/genbindings.js
+    ```
 
-  **NOTE:** We configure `getChild` should be controlled by module macro ``ENABLE_GET_CHILD` and it's disabled by default. So invoking `getChild` in JS will return `undefined` value.
+## Swig Interface File
 
-- Test c++ module configuration
+- There is a [swig-interface-template.i](swig-interface-template.i) in `engine/native/tools/swig-config` directory, just copy and rename it to some place in your project.
```

Comment:
The content is confused. Just copy and rename it. But don't know how to use it.

Reply:
Okay, I will fix it in the next Pull Request.